### ISSUE TITLE
False positive for SC2034; triggers when variable only used in "unset" command

### ISSUE THREAD

**Issue by slycordinator** [koalaman/shellcheck#3379]
#### For bugs with existing features

- Rule Id (if any, e.g. ): SC2034
- My shellcheck version (`shellcheck --version` or "online"): online
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

array=( foo bar baz )
del_index=1
unset 'array[del_index]'
for index in "${!array[@]}"; do
    echo "$index: ${array[$index]}"
done
```

#### Here's what shellcheck currently says:
Line 4:
del_index=1
^-- SC2034 (warning): del_index appears unused. Verify use (or export if used externally).

#### Here's what I wanted or expected to see:
(no output)

Note:
The erroneous warning is shown when I use either of:
`unset 'array[del_index]'`
`unset 'array[$del_index]'`

**Comment by nexushoratio** [comment]
As a work around, the following appears to both make shellcheck happy and function correctly:

```
unset "array[${del_index}]"                                                     |
```

That is, use double quotes and braces.
